src/libraries/System.Private.Xml.Linq/tests/misc/XAttribute.cs

```diff
@@ -17,5 +15,64 @@ public void FormattedDate()
             // Ensure we are compatible with the .NET Framework
             Assert.Equal("CreatedTime=\"2018-01-01T12:13:14Z\"", new XAttribute("CreatedTime", new DateTime(2018, 1, 1, 12, 13, 14, DateTimeKind.Utc)).ToString());
         }
+
+        public static IEnumerable<object[]> NumericValuesWithMinusSign()
+        {
+            yield return new object[] { -123 };
+            yield return new object[] { -123f };
+            yield return new object[] { -123L };
+            yield return new object[] { (short)-123 };
+            yield return new object[] { -12.3 };
+            yield return new object[] { -12.3m };
+            yield return new object[] { (sbyte)-123 };
+        }
+
+        [Theory]
+        [MemberData(nameof(NumericValuesWithMinusSign))]
+        public void MinusSignWithDifferentTypeSwedishCulture(object value)
+        {
+            try
+            {
+                Threading.Thread.CurrentThread.CurrentCulture = new CultureInfo("SE-SW");
+            }
+            catch (CultureNotFoundException) { /* Do nothing */ }
+
+            Assert.Equal('-', (new XAttribute("a", value)).Value[0]);
+        }
+
+        [Theory]
+        [MemberData(nameof(NumericValuesWithMinusSign))]
+        public void MinusSignWithDifferentTypeNoCulture(object value)
+        {
+            Assert.Equal('-', (new XAttribute("a", value)).Value[0]);
+        }
+
+        public static IEnumerable<object[]> NonNumericValues()
+        {
+            yield return new object[] { true, "true" };
+            yield return new object[] { new DateTimeOffset(2018, 1, 1, 12, 13, 14, TimeSpan.Zero), "2018-01-01T12:13:14Z" };
+            yield return new object[] { new TimeSpan(12, 13, 14), "PT12H13M14S" };
+            yield return new object[] { "-123\n", "-123\n" };
+        }
+
+        [Theory]
+        [MemberData(nameof(NonNumericValues))]
+        public void NonNumericTypeSwedishCulture(object value, string expected)
+        {
+            try
+            {
+                Threading.Thread.CurrentThread.CurrentCulture = new CultureInfo("SE-SW");
```

Comment:
We should reset the thread culture back to original so you don't affect the culture for the next test run on that thread. I think there is an IDisposable helper class for this somewhere.

Reply:
Yeah it's in ThreadCultureChange.cs.

Reply:
Good point, thanks!